azurelinuxagent/common/protocol/imds.py

```diff
@@ -11,6 +11,7 @@
 from azurelinuxagent.common.utils.flexible_version import FlexibleVersion
 
 IMDS_ENDPOINT = '169.254.169.254'
+IMDS_ENDPOINT_BACKUP = '168.63.129.16'
```

Comment:
The agent reads the Wireserver IP from DHCP settings. Never seen such a case (maybe azure stack uses something different) but maybe that would not be same as the one above. You should probably use the same uri that the agent is using instead of hardcoding it

Reply:
Is there some variable or class that I can call to get this DHCP value? Or do I need to call some class method to initiate the DHCP call and get the value?

Can you point me to what these are? Thanks!

Reply:
Hey Jack, you can use this function - 
```python
    def _get_wireserver_endpoint(self):
        try:
            file_path = os.path.join(conf.get_lib_dir(), ENDPOINT_FILE_NAME)
            return fileutil.read_file(file_path)
        except IOError as e:
            raise OSUtilError(ustr(e))
```

Reply:
We get it from DHCP and then write it to this file. Implementation - 
```python
        def _detect_wire_protocol(self):
        endpoint = self.dhcp_handler.endpoint
        if endpoint is None:
            '''
            Check if DHCP can be used to get the wire protocol endpoint
            '''
            (dhcp_available, conf_endpoint) =  self.osutil.is_dhcp_available()
            if dhcp_available:
                logger.info("WireServer endpoint is not found. Rerun dhcp handler")
                try:
                    self.dhcp_handler.run()
                except DhcpError as e:
                    raise ProtocolError(ustr(e))
                endpoint = self.dhcp_handler.endpoint
            else:
                logger.info("_detect_wire_protocol: DHCP not available")
                endpoint = self._get_wireserver_endpoint()
                if endpoint == None:
                    endpoint = conf_endpoint
                    logger.info("Using hardcoded WireServer endpoint {0}", endpoint)
                else:
                    logger.info("WireServer endpoint {0} read from file", endpoint)

        try:
            protocol = WireProtocol(endpoint)
            protocol.detect()
            self._set_wireserver_endpoint(endpoint)
            return protocol
        except ProtocolError as e:
            logger.info("WireServer is not responding. Reset endpoint")
            self.dhcp_handler.endpoint = None
            self.dhcp_handler.skip_cache = True
            raise e
```

Reply:
Very informative! I'll update my PR. Thanks Laveesh!

Reply:
This should be address